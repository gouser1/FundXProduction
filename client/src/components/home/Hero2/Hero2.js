import React, { useEffect, useState } from "react";
import axios from "axios";
import { Grid, Box, Button } from "@material-ui/core";
import useStyles from "./Hero2Style";
import { AuthContext } from "../../../helpers/AuthContext";

function Hero(props) {
  const { history } = props;
  const handleButtonClick = (pageURL) => {
    history.push(pageURL);
  };
  const classes = useStyles();

  const [authState, setAuthState] = useState(false);

  useEffect(() => {
    axios
      .get("http://localhost:3001/auth/auth", {
        headers: { accessToken: localStorage.getItem("accessToken") },
      })
      .then((response) => {
        if (response.data.error) {
          setAuthState(false);
        } else {
          setAuthState(true);
        }
      });
  }, []);

  return (
    <AuthContext.Provider value={{ authState, setAuthState }}>
      <div className={classes.root}>
        <Grid
          container
          spacing={0}
          direction="column"
          alignItems="center"
          justify="center"
          style={{ minHeight: "32vh" }}
        >
          <Grid item lg={6} md={10} sm={10} xs={12}>
            <Box textAlign="center">
              <h1 className={classes.h1}>
                Begin your investment adventure today!
              </h1>
            </Box>
          </Grid>
          <Grid item lg={12} md={10} sm={10} xs={12}>
            <Box textAlign="center">
              <h2 className={classes.h2}>
                Whether you are looking for investment or looking to invest.
                FundX has what you need.
              </h2>
            </Box>
          </Grid>
          <Grid item xs={12}>
            {!authState && (
              <>
                <Box textAlign="center" pt={2}>
                  <Button
                    onClick={() => handleButtonClick("/register")}
                    className={classes.button}
                  >
                    Find out more!
                  </Button>
                </Box>
              </>
            )}
            {authState && (
              <>
                <Box textAlign="center" pt={2}>
                  <Button
                    onClick={() => handleButtonClick("/dashboard/pitches")}
                    className={classes.button}
                  >
                    Find out more!
                  </Button>
                </Box>
              </>
            )}
          </Grid>
        </Grid>
      </div>
    </AuthContext.Provider>
  );
}

export default Hero;
