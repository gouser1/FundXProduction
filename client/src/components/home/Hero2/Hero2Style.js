import { makeStyles } from "@material-ui/styles";

export default makeStyles(() => ({
  root: {
    flexGrow: 1,
    backgroundColor: "#daf0e8",
    paddingBottom: "5vh",
  },
  typography: {
    fontFamily: "'Karla', sans-serif;",
    fontSize: "1.2em",
    color: "#46637f",
  },
  button: {
    fontFamily: "'Karla', sans-serif;",
    fontSize: "1.4em",
    color: "white",
    backgroundColor: "#3bc693",
    "&:hover": {
      backgroundColor: "#34b384",
    },
    textTransform: "none",
  },
  h1: {
    textAlign: "center",
    fontFamily: "'Karla', sans-serif;",
    fontSize: "1.5em",
    color: "#22303d",
    fontWeight: "700",
    "@media (min-width:700px)": {
      fontSize: "3rem",
    },
  },
  h2: {
    fontFamily: "'Rubik', sans-serif;",
    fontSize: "1.1em",
    color: "#46637f",
    "@media (min-width:700px)": {
      fontSize: "1.6rem",
    },
  },
}));
