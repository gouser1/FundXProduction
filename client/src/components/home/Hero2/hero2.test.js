import React from "react";
import renderer from "react-test-renderer";

import Hero2 from "./Hero2";

it("renders correctly", () => {
  const tree = renderer.create(<Hero2 />).toJSON();
  expect(tree).toMatchSnapshot();
});
