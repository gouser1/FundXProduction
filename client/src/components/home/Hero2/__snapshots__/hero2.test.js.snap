// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
    style={
      Object {
        "minHeight": "32vh",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-10 MuiGrid-grid-md-10 MuiGrid-grid-lg-6"
    >
      <div
        className="MuiBox-root MuiBox-root-6"
      >
        <h1
          className="makeStyles-h1-4"
        >
          Begin your investment adventure today!
        </h1>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-10 MuiGrid-grid-md-10 MuiGrid-grid-lg-12"
    >
      <div
        className="MuiBox-root MuiBox-root-7"
      >
        <h2
          className="makeStyles-h2-5"
        >
          Whether you are looking for investment or looking to invest. FundX has what you need.
        </h2>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiBox-root MuiBox-root-8"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-3"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Find out more!
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
